src/shutdown/umount.c

```diff
@@ -716,10 +732,49 @@ static int mount_points_list_umount(MountPoint **head, bool *changed, bool last_
                         continue;
 
                 /* Trying to umount */
-                if (umount_with_timeout(m, last_try) < 0)
+                r = umount_with_timeout(m, last_try);
+                if (r < 0)
                         n_failed++;
                 else
                         *changed = true;
+
+                /* If a mount is busy, we move it to not keep parent mount points busy.
+                 * If a mount point is not a leaf, moving it would invalidate our mount table.
+                 * More moving will occur in next iteration with a fresh mount table.
+                 *
+                 * umount_with_timeout will return EPROTO even on EBUSY due to forking.
+                 */
+                if (r != -EPROTO || !m->leaf)
+                        continue;
+
+                _cleanup_free_ char *dirname = NULL;
+
+                r = path_extract_directory(m->path, &dirname);
+                if (r < 0) {
+                        n_failed++;
+                        log_full_errno(last_try ? LOG_ERR : LOG_INFO, r, "Cannot find directory for %s: %m", m->path);
+                        continue;
+                }
+
+                if (!path_equal_or_files_same(dirname, "/run/shutdown/mounts", 0)) {
+                        char newpath[STRLEN("/run/shutdown/mounts/") + DECIMAL_STR_MAX(uint64_t)];
+
+                        xsprintf(newpath, "/run/shutdown/mounts/%08" PRIu64, random_u64());
+
+                        if (is_dir(m->path, true))
+                                (void) mkdir_p(newpath, 0700);
```

Comment:
i'd make the access mode 000 btw, it's not supposed to be accessed after all, the overmounted fs should be the only thing

Reply:
Done.